src/hyperion/external_interaction/callbacks/plan_reactive_callback.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Callable
+
+from bluesky.callbacks import CallbackBase
+from event_model.documents.event import Event
+
+if TYPE_CHECKING:
+    from event_model.documents.event import Event
+    from event_model.documents.event_descriptor import EventDescriptor
+    from event_model.documents.run_start import RunStart
+    from event_model.documents.run_stop import RunStop
+
+
+class PlanReactiveCallback(CallbackBase):
```

Comment:
Might be useful to have a brief comment explaining that this should be used when we only want callbacks while a plan is running